linera-chain/src/chain.rs

```diff
@@ -684,31 +708,7 @@ where
         #[cfg(with_metrics)]
         let _execution_latency = BLOCK_EXECUTION_LATENCY.measure_latency();
 
-        let chain_id = self.chain_id();
-        assert_eq!(block.chain_id, chain_id);
-        // The first incoming message of any child chain must be `OpenChain`. A root chain must
-        // already be initialized
-        if block.height == BlockHeight::ZERO
-            && self
-                .execution_state
-                .system
-                .description
-                .get()
-                .map_or(true, |description| description.is_child())
-        {
-            let (in_bundle, posted_message, config) = block
-                .starts_with_open_chain_message()
-                .ok_or_else(|| ChainError::InactiveChain(chain_id))?;
-            if !self.is_active() {
-                let message_id = MessageId {
-                    chain_id: in_bundle.origin.sender,
-                    height: in_bundle.bundle.height,
-                    index: posted_message.index,
-                };
-                self.execute_init_message(message_id, config, block.timestamp, local_time)
-                    .await?;
-            }
-        }
```

Comment:
it's very difficult to see if we indeed call `execute_init_message` prior to calling this `execute_block` function.

Reply:
It's called in `receive_message_bundle` [here](https://github.com/linera-io/linera-protocol/blob/a592336640d7d9d0ffd833fe99952fc533d1df2c/linera-chain/src/chain.rs#L504), so if we add an `OpenChain` message to an inbox, it is called, and we don't need to call it again when processing the first block.

If we don't see `OpenChain` before the first block _proposal_, we reject the proposal anyway.

If we don't see `OpenChain` before the first block and the block is already committed, then [we call it explicitly here](https://github.com/linera-io/linera-protocol/blob/a592336640d7d9d0ffd833fe99952fc533d1df2c/linera-core/src/chain_worker/state/attempted_changes.rs#L332) in `process_confirmed_block`, before executing the block.

So the call in `execute_block` was already redundant.